gpplot/plots.py

```diff
@@ -403,8 +389,65 @@ def dark_boxplot(data, x, y, boxprops=None, medianprops=None,
         capprops = {'color': 'black'}
     if flierprops is None:
         flierprops = {'marker': 'o', 'markerfacecolor': 'black'}
-    ax = sns.boxplot(data=data, x=x, y=y,
-                     boxprops=boxprops, medianprops=medianprops,
-                     whiskerprops=whiskerprops, capprops=capprops,
-                     flierprops=flierprops, **kwargs)
+    ax = sns.boxplot(data=data, x=x, y=y, boxprops=boxprops, medianprops=medianprops, whiskerprops=whiskerprops,
+                     capprops=capprops, flierprops=flierprops, **kwargs)
+    return ax
+
+
+def add_reg_line(data, x, y, ax=None, linestyle='dashed', linecolor='black', **kwargs):
+    """Add regression line to a scatter plot using seaborn.regplot
+
+    Parameters
+    ----------
+    data: DataFrame
+        DataFrame with columns x and y, same data used to create the scatter plot
+    x: str
+        x variable to regress
+    y: str
+        y variable to regress
+    ax: Axis object, optional
+        Plot to add regression line to
+    linestyle: str, optional
+        Style of regression line
+    linecolor: str, optional
+        Color of regression line
+    **kwargs
+        Other keyword arguments that are passed through to seaborn.regplot
+
+    Returns
+    -------
+    matplotlib.axes.Axes
+    """
+    ax = sns.regplot(data=data, x=x, y=y, scatter=False, ax=ax,
+                     line_kws={'linestyle': linestyle, 'color': linecolor}, **kwargs)
+    if ax is None:
```

Comment:
@mhegde I think this should come before sns.regplot, for when no axes are given to the function